tests/e2e/br/utils/s3/minio.go

```diff
@@ -0,0 +1,264 @@
+	ginkgo.By("init minio s3 storage")
+	pod := getMinioPod(ns)
+	if _, err := s.c.CoreV1().Pods(ns).Create(pod); err != nil {
+		return err
+	}
+	svc := getMinioService(ns)
+	if _, err := s.c.CoreV1().Services(ns).Create(svc); err != nil {
+		return err
+	}
+	secret := getMinioSecret(ns, accessKey, secretKey)
+	if _, err := s.c.CoreV1().Secrets(ns).Create(secret); err != nil {
+		return err
+	}
+	ginkgo.By("wait for minio s3 storage ready")
+
+	if err := podutil.WaitTimeoutForPodReadyInNamespace(s.c, minioName, ns, 5*time.Minute); err != nil {
+		return err
+	}
+
+	ep, err := s.forwardPort(ctx, ns)
+	if err != nil {
+		return err
+	}
+
+	mc, err := minio.New(ep, accessKey, secretKey, false)
+	if err != nil {
+		return err
+	}
+	if err = mc.MakeBucket(minioBucket, ""); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *minioStorage) forwardPort(ctx context.Context, ns string) (string, error) {
+	if s.fw == nil {
+		return getDefaultAddr(ns), nil
+	}
+	return portforward.ForwardOnePort(ctx, s.fw, ns, "svc/"+minioName, 9000)
+}
+
+func getDefaultAddr(ns string) string {
+	return fmt.Sprintf("%s.%s:9000", minioName, ns)
+}
+
+func (s *minioStorage) Config(ns, prefix string) *v1alpha1.S3StorageProvider {
+	return &v1alpha1.S3StorageProvider{
+		Provider:   v1alpha1.S3StorageProviderTypeCeph,
+		Prefix:     prefix,
+		SecretName: minioSecret,
+		Bucket:     minioBucket,
+		// scheme is necessary
+		Endpoint: "http://" + getDefaultAddr(ns),
+	}
+}
+
+// clean by deleting namespace, so just return
+func (s *minioStorage) Clean(ctx context.Context, ns string) error {
+	return nil
+}
+
+func (s *minioStorage) IsDataCleaned(ctx context.Context, ns, prefix string) (bool, error) {
+	accessKey, secretKey, err := s.accessSecret(ns)
+	if err != nil {
+		return false, err
+	}
+	ep, err := s.forwardPort(ctx, ns)
+	if err != nil {
+		return false, err
+	}
+	mc, err := minio.New(ep, accessKey, secretKey, false)
+	if err != nil {
+		return false, err
+	}
+	doneCh := make(chan struct{})
+	defer close(doneCh)
+	objs := mc.ListObjects(minioBucket, prefix, true, doneCh)
+	if len(objs) == 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (s *minioStorage) accessSecret(ns string) (string, string, error) {
+	secret, err := s.c.CoreV1().Secrets(ns).Get(minioSecret, metav1.GetOptions{})
+	if err != nil {
+		return "", "", err
+	}
+	accessKeyBytes, ok1 := secret.Data["access_key"]
+	secretKeyBytes, ok2 := secret.Data["secret_key"]
+	if !ok1 || !ok2 {
+		return "", "", fmt.Errorf("access_key or secret_key not found")
+	}
+	accessKey, err := base64DecodeToString(accessKeyBytes)
+	if err != nil {
+		return "", "", err
+	}
+	secretKey, err := base64DecodeToString(secretKeyBytes)
+	if err != nil {
+		return "", "", err
+	}
+
+	return accessKey, secretKey, nil
+}
+
+func base64DecodeToString(src []byte) (string, error) {
+	dst := make([]byte, 0, len(src))
+	if _, err := base64.StdEncoding.Decode(dst, src); err != nil {
+		return "", err
+	}
+	return string(dst), nil
```

Comment:
`dst` is always empty if reaching here

Reply:
Decode will fill content into dst

Reply:
I have retested. I'll fix it. Why CI passed...

Reply:
Fixed